HACKING.md

```diff
@@ -180,70 +184,58 @@ reason we create the app first is that otherwise OpenShift will
-needed once we are running on a newer version of OpenShift than 3.6 in
+The `jenkins:2` parameter is to match the tag name in the latest
+OpenShift. Some older versions of the template in OpenShift uses
+`jenkins:latest`. This will no longer be needed once we are running on a
+newer version of OpenShift than 3.6 in
 CentOS CI. See [#32](https://github.com/coreos/fedora-coreos-pipeline/pull/32)
 and [#70](https://github.com/coreos/fedora-coreos-pipeline/pull/70)
-for context on why this may be needed):
+for more context).
+
+### Creating the pipeline
+
+If working on the production pipeline, you may simply do:
 
 ```
-oc patch dc/jenkins -p '{"spec":{"triggers":[{"imageChangeParams":{"automatic":true,"containerNames":["jenkins"],"from":{"kind":"ImageStreamTag","name":"jenkins:2"}},"type":"ImageChange"},{"type":"ConfigChange"}]}}'
+./deploy --official --update
 ```
 
-### Create the pipeline from the template
+If working in a different namespace/cluster, the command is the same,
+but without the `--official` switch:
 
-If working on the production pipeline, you may simply do:
 
 ```
-oc new-app --file=manifests/pipeline.yaml
+./deploy --update
 ```
 
-If working on a local cluster you will want to override the defaults
-for some parameters:
-
-- `DEVEL_PREFIX` - **REQUIRED**
-    - The prefix to prepend to created resources.
-    - This is recommended to be your username followed by a dash.
-      i.e. `jlebon-`. The pipeline will simply not run without a
-      `DEVEL_PREFIX` set if it is not running in prod mode.
-- `PIPELINE_REPO_URL`
-    - Git source URI for pipeline Jenkinsfile
-- `PIPELINE_REPO_REF`
-    - Git branch/tag reference for pipeline Jenkinsfile
-- `PIPELINE_FCOS_CONFIG_URL`
-    - Git source URI for FCOS config
-- `PIPELINE_FCOS_CONFIG_REF`
-    - Git branch/tag reference for FCOS config
-- `PVC_SIZE`
-    - Size of the PVC to create
-- `COREOS_ASSEMBLER_IMAGE`
-    - Image of coreos-assembler to use
-- `S3_BUCKET`
-    - AWS S3 bucket in which to store builds (or blank for none)
+Developer-specific resources will be prefixed by the current user's
+username. You can provide `--prefix` to override this default.
 
+You may also want to provide additional switches depending on the
+circumstances. For example, to target specific pipeline, FCOS config,
+cosa image, or bucket:
 
 ```
-oc new-app --file=manifests/pipeline.yaml \
-    --param=DEVEL_PREFIX=$(whoami)- \
-    --param=PIPELINE_REPO_URL=https://github.com/jlebon/fedora-coreos-pipeline \
-    --param=PIPELINE_REPO_REF=my-feature-branch \
-    --param=S3_BUCKET=jlebon-fcos \
-    --param=PVC_SIZE=20Gi
+./deploy --update \
+    --pipeline https://github.com/jlebon/fedora-coreos-pipeline     \
+    --config https://github.com/jlebon/fedora-coreos-config@feature \
+    --cosa-img quay.io/jlebon/coreos-assembler:random-tag           \
+    --bucket jlebon-fcos
 ```
 
-This template creates:
+See `./deploy --help` for more information.
```

Comment:
I kind of liked having all of the parameters explained out here like we were doing before when we listed out all of the parameters of the template. Are you opposed to adding that stanza back but just convert it to reference arguments to deploy rather than template parameters?

Reply:
Done! :arrow_down: